File: Numerical_method/C++/lab3/lagrange.cpp
// Lagrange interpolation

#include<iostream>
using namespace std;
#include<conio.h>
int main()
{
	int n,i,j;
	float y[10],yp=0,xp,y0,y1,x[10],x0,x1,p;
	
	cout<<"Enter the number of terms :\t";
	cin>>n;
	cout<<"\n";
	for(i=0; i<n; i++)
	{
		
	cout<<"Enter the points  x"<<i+1<<" :\t";
	cin>>x[i];
	}
	cout<<"\n";
	for(i=0; i<n; i++)
	{
	cout<<"Enter the points  y"<<i+1<<" :\t";
	cin>>y[i];
	}
	cout<<"\nEnter the independent value xp :\t";
	cin>>xp;
	
	for(i=0; i<n; i++)
	{
		p=1;
		for(j=0; j<n; j++)
		{
		if(i!=j)
		{
			p=p*(xp-x[j])/(x[i]-x[j]);
		}
		
		}
		yp=yp+p*y[i];
	}
	cout<<"Interpolated value at \t"<<xp<<" is "<<yp;
	return 0;
}

File: Numerical_method/Python/lab3/lagrange.py
# Lagrange interpolation

def lagrange_interpolation(x_values, y_values, x_target):
    y_target = 0
    n = len(x_values)

    for i in range(n):
        p = 1
        for j in range(n):
            if i != j:
                p *= (x_target - x_values[j]) / (x_values[i] - x_values[j])
        y_target += p * y_values[i]

    return y_target

# Taking input from user
n = int(input("Enter the number of terms: "))

x_values = []
y_values = []

print("\nEnter the x values:")
for i in range(n):
    x = float(input(f"x{i + 1}: "))
    x_values.append(x)

print("\nEnter the y values:")
for i in range(n):
    y = float(input(f"y{i + 1}: "))
    y_values.append(y)

x_target = float(input("\nEnter the independent value x_target: "))

# Perform interpolation
y_target = lagrange_interpolation(x_values, y_values, x_target)
print(f"Interpolated value at {x_target} is {y_target:.5f}")
